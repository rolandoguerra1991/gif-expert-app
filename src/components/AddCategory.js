import React, { Fragment, useState } from "react";
import PropTypes from 'prop-types';

export default function AddCategory({ setCategories }) {

  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  }

  return (
    <Fragment>
      <form onSubmit={handleSubmit}>
        <input 
          type="text" 
          value={inputValue} 
          onChange={handleInputChange}
        />
      </form>
    </Fragment>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
