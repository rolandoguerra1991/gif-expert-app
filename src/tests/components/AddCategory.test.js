import AddCategory from '../../components/AddCategory.js'
import { shallow } from 'enzyme'


describe('Pruebas en <AddCateogry />', () => {
  const setCategories = jest.fn();
  const wrapper = shallow(<AddCategory setCategories={setCategories}/>);

  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    const event = { target: { value } };
    input.simulate('change', event);

    const inputValue = wrapper.find('input').prop('value');
    expect(inputValue).toBe(value);
  })
})
