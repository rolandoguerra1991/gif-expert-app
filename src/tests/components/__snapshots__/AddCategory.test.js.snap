// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCateogry /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1 />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
